lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,72 @@
+require 'vanagon/engine/base'
+require 'json'
+require 'lock_manager'
+
+LOCK_MANAGER_HOST = ENV['LOCK_MANAGER_HOST'] || 'redis'
+LOCK_MANAGER_PORT = ENV['LOCK_MANAGER_PORT'] || 6379
+VANAGON_LOCK_USER = "#{ENV['USER']}"
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        if @build_hosts.length == 1
+          @target = single_node_lock(@build_hosts[0])
+        else
+          @target = multi_node_lock(@build_hosts)
+        end
+      end
+
+      # Go through the lock process for a single node
+      def single_node_lock(host)
+        Vanagon::Driver.logger.info "Polling for a lock on #{host}."
+        @lockman.polling_lock(host, VANAGON_LOCK_USER, "Vanagon automated lock")
+        Vanagon::Driver.logger.info "Lock acquired on #{host}."
+        warn "Lock acquired on #{host} for #{VANAGON_LOCK_USER}."
```

Comment:
I think puts makes more sense to me, unless/until there is a clear breakdown of where messages go during a build